src/pages/RegisterPage.tsx

```diff
@@ -0,0 +1,119 @@
+import React, { useState, ChangeEvent, FormEvent, useEffect } from "react";
+import { useAuth } from "../hooks/useAuth";
+import { useNavigate } from "react-router-dom";
+
+const RegisterPage: React.FC = () => {
+  const [username, setUsername] = useState<string>("");
+  const [password, setPassword] = useState<string>("");
+  const [discord_username, setDiscord] = useState<string>("");
+  const [confirmPassword, setConfirmPassword] = useState<string>("");
+  const [error, setError] = useState<string>("");
+  const [successMessage, setSuccessMessage] = useState<string>("");
+
+  const { register, error: registerError, isAuthenticated } = useAuth();
+  const navigate = useNavigate(); 
+
+  useEffect(() => {
+    if (isAuthenticated) {
+        navigate("/");
+      }
+    }, [isAuthenticated, navigate]);    
+
+  const handleUserNameChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setUsername(event.target.value);
+  };
+
+  const handlePasswordChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setPassword(event.target.value);
+  };
+
+  const handleConfirmPasswordChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setConfirmPassword(event.target.value);
+  };
+
+  const handleDiscordChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setDiscord(event.target.value);
+  }
+
+  const handleRegister = async (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (password !== confirmPassword) {
+      setError("Passwords do not match.");
+      return;
+    }
+
+    await register(username, password, discord_username);
+    if (isAuthenticated) {
+      setSuccessMessage("Registration successful. You can now log in.");
+      setError("");
+    } else {
+      setError("Registration failed. Please try again.");
+    }
+  };
+
+  //TODO: Add messaging to tell people to type /verify "username" in the discord.
```

Comment:
Another TODO: figure out best flow for verification/redirection after. Don't want to open a two way channel, so could probably poll an endpoint to check if verified every ~5 seconds

Reply:
It will just be a message that says to type that then login after you've done it.

Reply:
So the issue is I'm not allowing the user to log in until the discord_id is verified so you cannot access the users data. If you want to change that you can, but then you're going to have to keep checking on the frontend which isn't as secure.

Reply:
The user record is created immediately, and with it their user id. If we return that immediately after account creation we can create a permission-less view to get verification status